src/components/decimal/decimal.test.tsx

```diff
@@ -0,0 +1,1406 @@
+      const user = userEvent.setup();
+      render(<Decimal defaultValue="1234.56" />);
+
+      const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+      decimalInput.focus();
+      decimalInput.setSelectionRange(selectionStart, selectionEnd);
+      await user.keyboard("{Delete}");
+
+      expect(decimalInput).toHaveValue(rawValue);
+      expect(screen.getByTestId("hidden-input")).toHaveValue(sanitisedValue);
+    }
+  );
+
+  // as above, a simple cursor placement (no selection) at the start or end need special handling
+  it("deletes the fisrt character when delete is pressed with the cursor at the start of the input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="1234.56" />);
+
+    const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+    decimalInput.focus();
+    await user.keyboard("{ArrowLeft}");
+    await user.keyboard("{Delete}");
+
+    expect(decimalInput).toHaveValue(",234.56");
+    expect(screen.getByTestId("hidden-input")).toHaveValue(",234.56");
+  });
+
+  it("doesn't delete anything when delete is pressed with the cursor at the end of the input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="1234.56" />);
+
+    const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+    decimalInput.focus();
+    await user.keyboard("{ArrowRight}");
+    await user.keyboard("{Delete}");
+
+    expect(decimalInput).toHaveValue("1,234.56");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("1234.56");
+  });
+
+  it.each([",", ".", "£", "$", "%", "@", "!", "*", "#"])(
+    "should not format multiple `%s` characters",
+    async (char) => {
+      const user = userEvent.setup();
+      render(<Decimal />);
+
+      const input = char.repeat(4);
+      await user.type(screen.getByRole("textbox"), input);
+      await user.tab();
+
+      expect(screen.getByRole("textbox")).toHaveValue(input);
+      expect(screen.getByTestId("hidden-input")).toHaveValue(input);
+    }
+  );
+
+  it("calls the onKeyPress callback when a key is pressed", async () => {
+    const user = userEvent.setup();
+    const onKeyPress = jest.fn();
+    render(<Decimal defaultValue="0.00" onKeyPress={onKeyPress} />);
+
+    screen.getByRole("textbox").focus();
+    await user.keyboard("{ArrowRight}");
+    await user.keyboard("1");
+
+    expect(onKeyPress).toHaveBeenCalledWith(
+      expect.objectContaining({ key: "1" })
+    );
+  });
+
+  it("has the correct data tag", () => {
+    render(<Decimal data-role="test-data-tag" />);
+    expect(screen.getByTestId("test-data-tag")).toHaveAttribute(
+      "data-component",
+      "decimal"
+    );
+  });
+
+  it("adds the expected HTML attributes to the hidden input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal name="example" />);
+
+    await user.type(screen.getByRole("textbox"), "1.23");
+    const hiddenInput = screen.getByTestId("hidden-input");
```

Comment:
comment: if we use `getByRole("textbox", { hidden: true })` we would get some of the below assertions implicitly

Reply:
[as noted above](https://github.com/Sage/carbon/pull/6905#discussion_r1724844170), this selector won't work